### PR TITLE
Mesh 1913/opt out of two way affirmations

### PR DESCRIPTION
**Pull request by HenriqueNogara**
This PR allows users to pre-affirm the receivement of transfers of certain tickers and custodians to pre-affirm the receivement of any transfers to a portfolio. 

### new features

- Adds the following extrinsics to the asset pallet: `exempt_ticker_affirmation`, `remove_ticker_affirmation_exemption`, `pre_approve_ticker`, `remove_ticker_pre_approval`;
- Adds the following extrinsics to the portfolio pallet: `pre_approve_portfolio`, `remove_portfolio_pre_approval`;

### modified logic

- `add_instruction` checks for pre-approved tickers/portfolios;

### CODE REVIEW

**Review by adamdossa — APPROVED** [review]

File: pallets/portfolio/src/lib.rs
Comment (Neopallium) [inline pallets/portfolio/src/lib.rs:871]: If we want both asset-level pre-approvals and portfolio-level pre-approvals.  Then we should check if the portfolio has a custodian (the owner has given custodianship control to another identity), before allowing asset-level pre-approvals.  Otherwise there is no point to restrict the portfolio-level pre-approvals management to the custodian, since the portfolio owner can bypass that restriction.

File: pallets/weights/src/pallet_asset.rs
Comment (Neopallium) [inline pallets/weights/src/pallet_asset.rs]: Please fill these in.  You can use the dev server to run the benchmarks if you don't want to run them locally.

File: pallets/weights/src/pallet_portfolio.rs
Comment (Neopallium) [inline pallets/weights/src/pallet_portfolio.rs]: Same here.

**Review by Neopallium — CHANGES_REQUESTED** [review]

**Review by Neopallium — APPROVED** [review]

**Pull request merged**
